Tests/SourceControlTests/GitRepositoryProviderTests.swift

```diff
@@ -40,4 +40,76 @@ class GitRepositoryProviderTests: XCTestCase {
             XCTAssertThrowsError(try provider.isValidDirectory(notGitChildPath))
         }
     }
+
+    func testIsValidDirectoryThrowsPrintableError() throws {
+        try testWithTemporaryDirectory { temp in
+            let provider = GitRepositoryProvider()
+            let expectedErrorMessage = "not a git repository"
+            do {
```

Comment:
question: out of curiosity, why did we not use `XCTAssertThrowsError` here?

https://developer.apple.com/documentation/xctest/1500795-xctassertthrowserror

Reply:
No good reason. Fixed in [8840c58](https://github.com/swiftlang/swift-package-manager/pull/8152/commits/8840c58ea6d14eb153ee01220e22c4763e2ac183).